stl/inc/mutex

```diff
@@ -31,37 +31,20 @@ _STD_BEGIN
 _EXPORT_STD class condition_variable;
 _EXPORT_STD class condition_variable_any;
 
-struct _Mtx_internal_imp_mirror {
-#ifdef _CRT_WINDOWS
-#ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 16;
-#else // _WIN64
-    static constexpr size_t _Critical_section_size = 8;
-#endif // _WIN64
-#else // _CRT_WINDOWS
-#ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 64;
-#else // _WIN64
-    static constexpr size_t _Critical_section_size = 36;
-#endif // _WIN64
-#endif // _CRT_WINDOWS
-
-    static constexpr size_t _Critical_section_align = alignof(void*);
-
-    int _Type;
-    _Aligned_storage_t<_Critical_section_size, _Critical_section_align> _Cs_storage;
-    long _Thread_id;
-    int _Count;
-};
-
-static_assert(sizeof(_Mtx_internal_imp_mirror) == _Mtx_internal_imp_size, "inconsistent size for mutex");
-static_assert(alignof(_Mtx_internal_imp_mirror) == _Mtx_internal_imp_alignment, "inconsistent alignment for mutex");
-
 class _Mutex_base { // base class for all mutex types
 public:
+#ifdef _USE_CONSTEXPR_MUTEX_CONSTRUCTOR
+    constexpr _Mutex_base(int _Flags = 0) noexcept {
+        _Mtx_storage._Critical_section = {};
+        _Mtx_storage._Thread_id        = -1;
+        _Mtx_storage._Type             = _Flags | _Mtx_try;
+        _Mtx_storage._Count            = 0;
+    }

```

Comment:
It seems to me that this definition is only well-formed since C++20 (WG21-P1331R2) (although I have no idea why MSVC and Clang accept this).
Likewise, changing the active member of a union is permitted in constant evaluation only since C++20 (WG21-P1330R0).

Perhaps the constant initialization can only be performed in C++20 mode and later due to the current implementation. Strangely, given `mutex` is not made a literal type and thus we can't define a `constexpr` `mutex` variable, IIUC we can't assert the constant initialization until C+20.

Reply:
`[[clang::require_constant_initialization]]` may be helpful.